src/h2_streams.erl

```diff
@@ -0,0 +1,541 @@
+-module(h2_streams).
+-include("http2.hrl").
+-include("h2_streams.hrl").
+
+-export(
+   [
+    new/1,
+    get/2,
+    close/4,
+    initiate_stream/7,
+    peer_initiated_stream/7,
+    replace/2,
+    send_what_we_can/4,
+    sort/1,
+    notify_pid/1,
+    update_all_recv_windows/2,
+    update_all_send_windows/2,
+    update_peer_max_active/2,
+    update_self_max_active/2
+   ]).
+
+-spec new(
+        client | server
+       ) -> streams().
+new(client) ->
```

Comment:
could this be something like `initialise` or `setup`? h2_streams:new feels like it should make a new stream